client/webserver/site/src/js/notifications.ts

```diff
@@ -35,79 +35,115 @@ const NoteTypeMatch = 'match'
 }
 
-export const defaultBrowserNtfnSettings: BrowserNtfnSetting = {
+export const defaultDesktopNtfnSettings: DesktopNtfnSetting = {
   [NoteTypeOrder]: true,
   [NoteTypeMatch]: true,
   [NoteTypeBondPost]: true,
   [NoteTypeConnEvent]: true
 }
 
-let browserNtfnSettings: BrowserNtfnSetting
+let desktopNtfnSettings: DesktopNtfnSetting
 
-export function ntfnPermissionGranted () {
-  return window.Notification.permission === 'granted'
-}
+// BrowserNotifier is a wrapper around the browser's notification API.
+class BrowserNotifier {
+  static ntfnPermissionGranted (): boolean {
+    return window.Notification.permission === 'granted'
+  }
 
-export function ntfnPermissionDenied () {
-  return window.Notification.permission === 'denied'
-}
+  static ntfnPermissionDenied (): boolean {
+    return window.Notification.permission === 'denied'
+  }
 
-export async function requestNtfnPermission () {
-  if (!('Notification' in window)) {
-    return
+  static async requestNtfnPermission (): Promise<void> {
+    if (!('Notification' in window)) {
+      return
+    }
+    if (BrowserNotifier.ntfnPermissionGranted()) {
+      BrowserNotifier.sendDesktopNotification(intl.prep(intl.ID_BROWSER_NTFN_ENABLED))
+    } else if (!BrowserNotifier.ntfnPermissionDenied()) {
+      await Notification.requestPermission()
+      BrowserNotifier.sendDesktopNotification(intl.prep(intl.ID_BROWSER_NTFN_ENABLED))
+    }
   }
-  if (Notification.permission === 'granted') {
-    showBrowserNtfn(intl.prep(intl.ID_BROWSER_NTFN_ENABLED))
-  } else if (Notification.permission !== 'denied') {
-    await Notification.requestPermission()
-    showBrowserNtfn(intl.prep(intl.ID_BROWSER_NTFN_ENABLED))
+
+  static sendDesktopNotification (title: string, body?: string) {
+    if (window.Notification.permission !== 'granted') return
+    const ntfn = new window.Notification(title, {
+      body: body,
+      icon: '/img/softened-icon.png'
+    })
+    return ntfn
   }
 }
 
-export function showBrowserNtfn (title: string, body?: string) {
-  if (window.Notification.permission !== 'granted') return
-  const ntfn = new window.Notification(title, {
-    body: body,
-    icon: '/img/softened-icon.png'
-  })
-  return ntfn
+// OSDesktopNotifier manages OS desktop notifications via the same interface
+// as BrowserNotifier, but sends notifications using an underlying Go
+// notification library exposed to the webview.
+class OSDesktopNotifier {
+  static ntfnPermissionGranted (): boolean {
+    return true
+  }
+
+  static ntfnPermissionDenied (): boolean {
+    return false
+  }
+
+  static async requestNtfnPermission (): Promise<void> {
+    OSDesktopNotifier.sendDesktopNotification(intl.prep(intl.ID_BROWSER_NTFN_ENABLED))
+    return Promise.resolve()
+  }
+
+  static sendDesktopNotification (title: string, body?: string): void {
+    // this calls a function exported via webview.Bind()
+    const w = (window as any)
+    w.sendOSNotification(title, body)
```

Comment:
And we still need to check permissions here before attempting to send a notification.